docs/featuretoggles/extensions/featuretoggles.py

```diff
@@ -0,0 +1,98 @@
+import os
+
+from docutils import nodes
+from docutils.parsers.rst import Directive, directives
+import yaml
+
+REPORT_PATH = os.path.join(os.path.dirname(__file__), "..", "report.yml")
+
+
+def load_feature_toggles(report_path):
+    """ Load the feature toggles listed in the report file.
+    Return:
+        toggles (dict): feature toggles indexed by name.
+    """
+    with open(report_path) as report_file:
+        report = yaml.safe_load(report_file)
+
+    group_id_names = {}
+    toggles = {}
+    for entries in report.values():
+        for entry in entries:
+            group_id = entry["report_group_id"]
+            key = entry["annotation_token"]
+            value = entry["annotation_data"]
+            if key == ".. toggle_name:":
+                toggle_name = value
+                group_id_names[group_id] = toggle_name
+                toggles[toggle_name] = {
+                    "filename": entry["filename"],
+                    "line_number": entry["line_number"],
+                }
+            else:
+                toggles[group_id_names[group_id]][key] = value
+
+    return toggles
+
+
+class FeatureToggles(Directive):
+    """
+    This class requires the presence of a report.yml file in the parent folder.
+
+    Use this directive as follows::
+
+        .. featuretoggles::
+            :github_repo: https://github.com/edx/edx-platform
+            :repo_version: open-release/zealot.4
+
+    TODO: improve this docstring
+    """
+
+    REPORT_PATH = os.path.join(os.path.dirname(__file__), "..", "report.yml")
+    required_arguments = 0
+    optional_arguments = 2
+    option_spec = {
+        "github_repo": directives.unchanged_required,
+        "repo_version": directives.unchanged,
+    }
+
+    def run(self):
+        toggle_nodes = list(self.iter_nodes())
+        return [nodes.section("", *toggle_nodes, ids=["featuretoggles"])]
+
+    def iter_nodes(self):
+        toggles = load_feature_toggles(self.REPORT_PATH)
+        for toggle_name in sorted(toggles):
+            toggle = toggles[toggle_name]
+            yield nodes.title(text=toggle_name)
+            yield nodes.paragraph(
+                text="Default: {}".format(
+                    toggle.get(".. toggle_default:", "Not defined")
+                )
+            )
+            yield nodes.paragraph(
+                "",
+                "Source: ",
+                nodes.reference(
+                    text="{} (line {})".format(
+                        toggle["filename"], toggle["line_number"]
+                    ),
+                    refuri="{}/blob/{}/{}#L{}".format(
+                        self.options.get("github_repo"),
+                        self.options.get("repo_version", "master"),
```

Comment:
Note: If these urls use the latest hash on master instead, they wouldn't break in the future as master changes.

Reply:
Yes, the plan is to use the current sha1 from the git repo (see the TODO in the [index.rst](https://github.com/edx/edx-platform/pull/24344/files#diff-2119163249c3344e0bf5dc9a06c2376d) file)